fearless/Modules/Staking/StakingPayoutConfirmation/StakingPayoutConfirmationInteractor.swift

```diff
@@ -84,236 +86,277 @@ final class StakingPayoutConfirmationInteractor {
-            deliverOn: .main,
-            executing: updateClosure,
-            failing: failureClosure,
-            options: options
+        presenter.didRecieve(
+            account: account,
+            rewardAmount: rewardAmount
         )
     }
 
-    private func subscribeToPriceChanges() {
-        let updateClosure = { [weak self] (changes: [DataProviderChange<PriceData>]) in
-            if changes.isEmpty {
-                self?.presenter.didReceive(price: nil)
-            } else {
-                for change in changes {
-                    switch change {
-                    case let .insert(item), let .update(item):
-                        self?.presenter.didReceive(price: item)
-                    case .delete:
-                        self?.presenter.didReceive(price: nil)
-                    }
-                }
-            }
-        }
+    private func processBatches() {
+        guard let firstPayout = payouts.first else { return }
 
-        let failureClosure = { [weak self] (error: Error) in
-            self?.presenter.didReceive(priceError: error)
+        guard payouts.count > 1 else {
+            batches = [payouts]
+            estimateFee()
             return
         }
 
-        let options = DataProviderObserverOptions(
-            alwaysNotifyOnRefresh: false,
-            waitsInProgressSyncOnAdd: false
-        )
-
-        priceProvider.addObserver(
-            self,
-            deliverOn: .main,
-            executing: updateClosure,
-            failing: failureClosure,
-            options: options
-        )
+        estimateFee(for: [firstPayout], with: FeeType.elaborateBlockWeight.rawValue)
     }
 
-    private func subscribeToStashControllerProvider() {
-        guard stashControllerProvider == nil, let selectedAccount = settings.selectedAccount else {
+    private func estimateFee(for batch: Batch, with identifier: String) {
+        guard let closure = createExtrinsicBuilderClosure(for: batch) else {
+            presenter.didReceiveFee(result: .failure(CommonError.undefined))
             return
         }
 
-        let provider = substrateProviderFactory.createStashItemProvider(for: selectedAccount.address)
+        feeProxy.estimateFee(
+            using: extrinsicService,
+            reuseIdentifier: identifier,
+            setupBy: closure
+        )
+    }
+}
 
-        let changesClosure: ([DataProviderChange<StashItem>]) -> Void = { [weak self] changes in
-            let stashItem = changes.reduceToLastChange()
-            self?.handle(stashItem: stashItem)
-        }
+// MARK: - StakingPayoutConfirmationInteractorInputProtocol
 
-        let failureClosure: (Error) -> Void = { [weak self] error in
-            self?.presenter.didReceive(stashItemError: error)
+extension StakingPayoutConfirmationInteractor: StakingPayoutConfirmationInteractorInputProtocol {
+    func setup() {
+        guard let selectedAccountAddress = settings.selectedAccount?.address else {
```

Comment:
Is there any use case for settings other then fetching selected account? It might be simpler to pass the account address directly to interactor